overlord/snapstate/snapstate.go

```diff
@@ -823,8 +828,13 @@ func InstallMany(st *state.State, names []string, userID int) ([]string, []*stat
 			return nil, nil, err
 		}
 
+		channel := "stable"
+		if sar.RedirectChannel != "" {
+			channel = sar.RedirectChannel
```

Comment:
is this bit unit tested?

Reply:
I added an InstallMany unit test that checks this.